Content.Client/_Harmony/JoinQueue/QueueState.cs

```diff
@@ -0,0 +1,54 @@
+using Content.Shared._Harmony.Common.JoinQueue;
+using Robust.Client.Audio;
+using Robust.Client.Console;
+using Robust.Client.State;
+using Robust.Client.UserInterface;
+using Robust.Shared.Player;
+
+namespace Content.Client._Harmony.JoinQueue;
+
+public sealed class QueueState : State
+{
+    [Dependency] private readonly IUserInterfaceManager _userInterface = default!;
+    [Dependency] private readonly IClientConsoleHost _console = default!;
+
+
+    private const string JoinSoundPath = "/Audio/Effects/newplayerping.ogg";
+
+    private QueueGui? _gui;
+
+
+    protected override void Startup()
+    {
+        _gui = new QueueGui();
+        _userInterface.StateRoot.AddChild(_gui);
+
+        _gui.QuitPressed += OnQuitPressed;
+    }
+
+    protected override void Shutdown()
+    {
+        _gui!.QuitPressed -= OnQuitPressed;
+        _gui.Dispose();
```

Comment:
Disposing a control is deprecated
```suggestion
        _userInterface.StateRoot.RemoveChild(_gui);
```